pkg/envoy/xds/ack.go

```diff
@@ -245,6 +272,18 @@ func (m *AckingResourceMutatorWrapper) Delete(typeURL string, resourceName strin
 	m.locker.Lock()
 	defer m.locker.Unlock()
 
+	wait := wg != nil
+
+	if m.restoring {
```

Comment:
The restore flag seems to be for all  resources but not only for NP, is it intentional to simplify implementation ?

Reply:
Yes, the change in xds/server.go does not need to be specific to a resource type. This way all the resource types benefit from reduced churn on agent restart.